src/stratigraphy/util/predictions.py

```diff
@@ -265,31 +162,232 @@ def from_json(cls, prediction_from_file: dict) -> "OverallFilePredictions":
         overall_file_predictions = OverallFilePredictions()
         for file_name, file_data in prediction_from_file.items():
             metadata = BoreholeMetadata.from_json(file_data["metadata"], file_name)
-            layers = LayerPrediction.from_json(file_data["layers"])
+
+            layers = Layer.from_json(file_data["layers"])
+            layers_on_page = LayersOnPage(layers_on_page=layers)
+            layers_in_doc = LayersInDocument(
+                layers_in_document=[layers_on_page], filename=file_name
+            )  # TODO: This is a bit of a hack as we do not seem to save the page of the layer
+
             depths_materials_columns_pairs = [
                 DepthsMaterialsColumnPairs.from_json(dmc_pair)
                 for dmc_pair in file_data["depths_materials_column_pairs"]
             ]
-            groundwater_entries = [GroundwaterInformationOnPage.from_json(entry) for entry in file_data["groundwater"]]
+
+            groundwater_entries = [GroundwaterOnPage.from_json(entry) for entry in file_data["groundwater"]]
+            groundwater_in_document = GroundwaterInDocument(groundwater=groundwater_entries, filename=file_name)
             overall_file_predictions.add_file_predictions(
                 FilePredictions(
-                    layers=layers,
+                    layers=layers_in_doc,
                     file_name=file_name,
                     metadata=metadata,
                     depths_materials_columns_pairs=depths_materials_columns_pairs,
-                    groundwater_entries=groundwater_entries,
+                    groundwater=groundwater_in_document,
                 )
             )
         return overall_file_predictions
 
+    def get_groundwater_entries(self) -> list[GroundwaterInDocument]:
```

Comment:
is there any value in defining a dedicated method for this one-liner?

Reply:
I initially thought that it would make the code a bit clearer. However, since we are not doing any processing, I am happy to remove that function. I do believe your code in branch `LGVISIUM-80/review-stijn` already fixes this.